src/util/Iterators.h

```diff
@@ -260,6 +260,9 @@ class InputRangeMixin {
   Sentinel end() const { return {}; }
 };
 
+// No details empty struct, the default for no details
+struct NoDetails {};
```

Comment:
We have too many `NoDetails` types.... unify them (there should be exactly one, the others should be either non-existent or just `using` aliases.